cranelift/codegen/src/isa/x64/inst/emit.rs

```diff
@@ -1584,6 +1585,55 @@ pub(crate) fn emit(
             state.adjust_virtual_sp_offset(-callee_pop_size);
         }
 
+        Inst::ReturnCallKnown {
+            callee,
+            info: call_info,
+        } => {
+            emit_return_call_common_sequence(
+                allocs,
+                sink,
+                info,
+                state,
+                call_info.new_stack_arg_size,
+                call_info.old_stack_arg_size,
+                call_info.ret_addr,
+                call_info.fp,
+                call_info.tmp,
+                &call_info.uses,
+            );
+
+            // Finally, jump to the callee!
```

Comment:
A comment here noting why not delegate to `Inst::JmpKnown` (in the usual pseudoinst way, i.e. `let inst = ...; inst.emit(...);`) would be good. I think the main reason is that we have different metadata in this case, in particular we don't have a label for the target but rather a different kind of relocation; is that right?

Reply:
That's correct. Added a comment.